addons/pos_sale/static/src/js/OrderManagementScreen/SaleOrderManagementScreen.js

```diff
@@ -291,7 +291,8 @@ odoo.define('pos_sale.SaleOrderManagementScreen', function (require) {
             sale_order[0].amount_unpaid = saleOrdersAmountUnpaid[sale_order[0].id];
 
             const sale_lines = await this._getSOLines(sale_order[0].order_line);
-            sale_order[0].order_line = sale_lines;
+            const promo_products_to_remove = this.env.pos.promo_programs ? this.env.pos.promo_programs.flatMap(program => program.promo_code_usage === 'no_code_needed' ? program.discount_line_product_id[0] : []) : [];
```

Comment:
I added this to make sure you can settle order when pos_coupon is not installed. If not `promo_programs.flatMap()` would return an error because `promo_programs` was undefined